massa-wallet/src/lib.rs

```diff
@@ -40,15 +37,16 @@ pub struct Wallet {
 impl Wallet {
     /// Generates a new wallet initialized with the provided file content
     pub fn new(path: PathBuf, password: String) -> Result<Wallet, WalletError> {
-        let content = std::fs::read(&path)?;
+        let content = &std::fs::read(&path)?[..];
         let priv_keys = if !content.is_empty() {
             // If wallet contains data decipher it
             let cipher = Aes256GcmSiv::new(Key::from_slice(
                 Hash::compute_from(password.as_bytes()).to_bytes(),
             ));
-            let nonce = Nonce::from_slice(NONCE_SLICE);
+            let mut iter = content.split(|c| c == &b':');
```

Comment:
no need for a separator since the nonce is always 12 bytes

You can just set NONCE_SIZE = 12 then use it as `let nonce = Nonce::from_slice(content.get(0..NONCE_SIZE)?);`

and then `decrypt(content.get([NONCE_SIZE..])?)`